src/sections/channels.tsx

```diff
@@ -0,0 +1,492 @@
+    const tabLabel = tabIndex === 0 ? 'incoming' : 'outgoing';
+    const channelsData =
+      tabIndex === 0 ? channels?.incoming : channels?.outgoing;
+
+    return (
+      <button
+        disabled={!channelsData || Object.keys(channelsData).length === 0}
+        onClick={() => {
+          if (channelsData) {
+            exportToCsv(
+              Object.entries(channelsData).map(([, channel]) => ({
+                channelId: channel.channelId,
+                peerId: channel.peerId,
+                status: channel.status,
+                dedicatedFunds: channel.balance,
+              })),
+              `${tabLabel}-channels.csv`
+            );
+          }
+        }}
+      >
+        export
+      </button>
+    );
+  };
+
+  const handleCloseChannels = (
+    direction: 'incoming' | 'outgoing',
+    peerId: string,
+    channelId: string
+  ) => {
+    set_closingStates((prevStates) => ({
+      ...prevStates,
+      [channelId]: {
+        closing: true,
+        closeSuccess: false,
+        closeErrors: [],
+      },
+    }));
+
+    dispatch(
+      actionsAsync.closeChannelThunk({
+        apiEndpoint: loginData.apiEndpoint!,
+        apiToken: loginData.apiToken!,
+        direction: direction,
+        peerId: peerId,
+      })
+    )
+      .unwrap()
+      .then(() => {
+        set_closingStates((prevStates) => ({
+          ...prevStates,
+          [channelId]: {
+            closing: false,
+            closeSuccess: true,
+            closeErrors: [],
+          },
+        }));
+        handleRefresh();
+      })
+      .catch((e) => {
+        set_closingStates((prevStates) => ({
+          ...prevStates,
+          [channelId]: {
+            closing: false,
+            closeSuccess: false,
+            closeErrors: [
+              ...(prevStates[channelId]?.closeErrors || []),
+              {
+                error: e.error,
+                status: e.status,
+              },
+            ],
+          },
+        }));
+      });
+  };
+
+  const fundChannel = (channelId: string, peerId: string) => {
+    return (
+      <>
+        <button onClick={() => handleOpenFundingPopup(channelId)}>Fund</button>
+        <Dialog
+          open={openFundingPopups[channelId] || false}
+          onClose={() => closeFundingPopup(channelId)}
+        >
+          <DialogTitle>Fund Channel</DialogTitle>
+          <DialogContent>
+            <TextField
+              label="Funding Amount"
+              type="number"
+              value={fundingAmount}
+              onChange={(e) => set_fundingAmount(e.target.value)}
+            />
+          </DialogContent>
+          <DialogActions>
+            <button onClick={() => closeFundingPopup(channelId)}>Cancel</button>
+            <button
+              onClick={() => handleFundChannels(peerId, channelId)}
+              disabled={!fundingAmount || parseFloat(fundingAmount) <= 0}
+            >
+              Fund
+            </button>
```

Comment:
After clicking, you need to disable the button and show loader, so we know it was clicked or just close the popup right away. Lets go with the 2nd option.